components/frontend/src/App.js

```diff
@@ -134,20 +143,28 @@ class App extends Component {
         history.push(target + (search.length > 0 ? "?" + search : ""));
     }
 
-    connect_to_nr_measurements_event_source() {
+    connectToNrMeasurementsEventSource() {
         this.source = new EventSource(nr_measurements_api);
         let self = this;
-        this.source.addEventListener('init', function (e) {
-            self.setState({ nr_measurements: Number(e.data) });
-        }, false);
-        this.source.addEventListener('delta', function (e) {
-            self.setState({ nr_measurements: Number(e.data) }, () => self.reload());
-        }, false);
-        this.source.addEventListener('error', function (e) {
-            if (e.readyState === EventSource.CLOSED || e.readyState === EventSource.OPEN) {
-                self.setState({ nr_measurements: 0 });
+        this.source.addEventListener("init", (message) => {
+            const newNrMeasurements = Number(message.data);
+            if (!self.state.nrMeasurementsStreamConnected) {
+                showMessage("success", "Connected to server", "Successfully reconnected to server.")
```

Comment:
Is there a risk of toast-spam when a connection is (very) unstable?

Reply:
The timeout for the reconnection attempt is 20 seconds, which is about as long as the toasters are visible, so that matches.